components/org.wso2.carbon.identity.query.saml/src/main/java/org/wso2/carbon/identity/query/saml/handler/SAMLAssertionFinderImpl.java

```diff
@@ -58,7 +60,8 @@ public void init() {
      * @throws  IdentitySAML2QueryException If request message not contain AssertionId
      */
     public Assertion findByID(String id) throws IdentitySAML2QueryException {
-        Assertion assertion = null;
+
+        Assertion assertion;
         if (id.length() > 0) {
```

Comment:
this is a public method - can id be null?

Reply:
Fixed